kraft-shade/src/main/java/com/cardinalblue/kraftshade/shader/KraftShader.kt

```diff
@@ -42,6 +44,21 @@ abstract class KraftShader : SuspendAutoCloseable {
 
     protected val logger = KraftLogger(this::class.simpleName ?: "KraftShader")
 
+    open fun loadVertexShader(): String {
+        return DEFAULT_VERTEX_SHADER
+    }
+
+    abstract fun loadFragmentShader(): String
+
+    /**
+     *  This method is used to intercept the fragment shader source code before it is compiled.
+     *  For example, you can add OES extension to the fragment shader if the input texture is an
+     *  [ExternalOESTexture].
+     */
+    open fun interceptFragmentShader(fragmentShader: String): String {
```

Comment:
suggest:
Replacing string in fragment code is quite intrusive to me, so as discussed, I prefer using a single shader that has OES extension and convert the video texture into a temporary buffer for further process.